File: wp-post.py
#!/usr/bin/env python3
"""
WordPress REST API ile otomatik makale gönderme.
Scraper çıktısını WordPress'e otomatik yayınlar.

Kullanım:
  python3 wp-post.py

Önce config.json dosyasını düzenleyin.
WordPress → Kullanıcılar → Profiliniz → Uygulama Şifreleri bölümünden
yeni bir uygulama şifresi oluşturun.
"""

import os
import json
import requests
from datetime import datetime, timezone, timedelta

TR_TZ = timezone(timedelta(hours=3))


def load_config():
    """config.json dosyasını yükler."""
    config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "config.json")
    
    if not os.path.exists(config_path):
        # Örnek config oluştur
        example_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "config.example.json")
        print(f"❌ config.json bulunamadı!")
        print(f"   config.example.json dosyasını config.json olarak kopyalayıp düzenleyin:")
        print(f"   cp config.example.json config.json")
        return None
    
    with open(config_path, "r", encoding="utf-8") as f:
        return json.load(f)


def post_to_wordpress(html_content, title=None):
    """HTML içeriğini WordPress'e gönderi olarak yayınlar."""
    config = load_config()
    if not config:
        return False

    wp = config["wordpress"]
    now = datetime.now(TR_TZ)
    
    if not title:
        title = f"Günlük Kamu Rehberi - {now.strftime('%d.%m.%Y')}"

    # WordPress REST API endpoint
    api_url = f"{wp['url'].rstrip('/')}/wp-json/wp/v2/posts"

    post_data = {
        "title": title,
        "content": html_content,
        "status": "publish",
        "categories": [wp.get("category_id", 1)],
        "meta": {
            "_yoast_wpseo_metadesc": f"Bugünün Resmi Gazete kararları ve İŞKUR ilanları - {now.strftime('%d.%m.%Y')}"
        }
    }

    try:
        response = requests.post(
            api_url,
            json=post_data,
            auth=(wp["username"], wp["app_password"]),
            headers={"Content-Type": "application/json"},
            timeout=30
        )

        if response.status_code == 201:
            post = response.json()
            print(f"✅ WordPress'e yayınlandı!")
            print(f"   URL: {post.get('link', 'N/A')}")
            print(f"   ID: {post.get('id', 'N/A')}")
            return True
        else:
            print(f"❌ WordPress hatası: {response.status_code}")
            print(f"   Yanıt: {response.text[:200]}")
            return False

    except Exception as e:
        print(f"❌ Bağlantı hatası: {e}")
        return False


def main():
    """Scraper çıktısını okuyup WordPress'e gönderir."""
    output_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "output", "index.html")
    
    if not os.path.exists(output_path):
        print("❌ output/index.html bulunamadı. Önce scraper.py çalıştırın.")
        print("   python3 scraper.py")
        return

    with open(output_path, "r", encoding="utf-8") as f:
        html_content = f.read()

    # Body içeriğini çıkar (WordPress için sadece body kısmı)
    from bs4 import BeautifulSoup
    soup = BeautifulSoup(html_content, "lxml")
    body = soup.find("body")
    
    if body:
        content = str(body.decode_contents())
    else:
        content = html_content

    print("📤 WordPress'e gönderiliyor...")
    post_to_wordpress(content)


if __name__ == "__main__":
    main()

File: wp-post.js
#!/usr/bin/env node
/**
 * WordPress REST API ile otomatik makale gönderme (Node.js versiyonu).
 * 
 * Kullanım:
 *   node wp-post.js
 * 
 * Önce config.json dosyasını düzenleyin.
 */

const fs = require('fs');
const path = require('path');
const https = require('https');
const http = require('http');

function loadConfig() {
  const configPath = path.join(__dirname, 'config.json');
  
  if (!fs.existsSync(configPath)) {
    console.error('❌ config.json bulunamadı!');
    console.error('   config.example.json dosyasını config.json olarak kopyalayıp düzenleyin:');
    console.error('   cp config.example.json config.json');
    return null;
  }
  
  return JSON.parse(fs.readFileSync(configPath, 'utf-8'));
}

function postToWordPress(htmlContent, title) {
  return new Promise((resolve, reject) => {
    const config = loadConfig();
    if (!config) { reject(new Error('Config yok')); return; }
    
    const wp = config.wordpress;
    const now = new Date();
    const dateStr = now.toLocaleDateString('tr-TR', { timeZone: 'Europe/Istanbul' });
    
    if (!title) {
      title = `Günlük Kamu Rehberi - ${dateStr}`;
    }
    
    const postData = JSON.stringify({
      title: title,
      content: htmlContent,
      status: 'publish',
      categories: [wp.category_id || 1]
    });
    
    const url = new URL(`${wp.url.replace(/\/$/, '')}/wp-json/wp/v2/posts`);
    const auth = Buffer.from(`${wp.username}:${wp.app_password}`).toString('base64');
    
    const options = {
      hostname: url.hostname,
      port: url.port || (url.protocol === 'https:' ? 443 : 80),
      path: url.pathname,
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Basic ${auth}`,
        'Content-Length': Buffer.byteLength(postData)
      }
    };
    
    const client = url.protocol === 'https:' ? https : http;
    
    const req = client.request(options, (res) => {
      let data = '';
      res.on('data', chunk => { data += chunk; });
      res.on('end', () => {
        if (res.statusCode === 201) {
          const post = JSON.parse(data);
          console.log('✅ WordPress\'e yayınlandı!');
          console.log(`   URL: ${post.link || 'N/A'}`);
          console.log(`   ID: ${post.id || 'N/A'}`);
          resolve(true);
        } else {
          console.error(`❌ WordPress hatası: ${res.statusCode}`);
          console.error(`   Yanıt: ${data.substring(0, 200)}`);
          resolve(false);
        }
      });
    });
    
    req.on('error', (e) => {
      console.error(`❌ Bağlantı hatası: ${e.message}`);
      reject(e);
    });
    
    req.write(postData);
    req.end();
  });
}

async function main() {
  const outputPath = path.join(__dirname, 'output', 'index.html');
  
  if (!fs.existsSync(outputPath)) {
    console.error('❌ output/index.html bulunamadı. Önce scraper çalıştırın.');
    return;
  }
  
  const htmlContent = fs.readFileSync(outputPath, 'utf-8');
  
  // Body içeriğini çıkar
  const bodyMatch = htmlContent.match(/<body[^>]*>([\s\S]*)<\/body>/i);
  const content = bodyMatch ? bodyMatch[1] : htmlContent;
  
  console.log('📤 WordPress\'e gönderiliyor...');
  await postToWordPress(content);
}

main().catch(console.error);